hazelcast/src/main/java/com/hazelcast/jet/impl/MasterJobContext.java

```diff
@@ -582,6 +582,12 @@ private void onStartExecutionComplete(Throwable error, Collection<Entry<MemberIn
                     // have to use Async version, the future is completed inside a synchronized block
                     .whenCompleteAsync(withTryCatch(logger, (r, e) -> finalizeJob(finalError)));
         } else {
+            if (error instanceof ExecutionNotFoundException) {
+                // If the StartExecutionOperation didn't find the execution, it means that we must have cancelled it.
```

Comment:
we rely on assertions to check if the job is cancelled

Reply:
because I assume that every time this happens, the the termination of execution without a snapshot must have happened.

Reply:
I merged this PR without resolving this comment because the test failed another time in my PR. We can still discuss this when you're back.